feat: Add basic authentication (#1005)
DSD-DBS/capella-collab-manager: docs/user/docs/tokens.md

<!--
 ~ SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
 ~ SPDX-License-Identifier: Apache-2.0
 -->

# Token Authentication

To authenticate against the API you can either use a Bearer token (which the
browser usually uses) or authenticate with a longer lived basic authentication

> token which can be used e.g. in scripts.

Reviewer: ```suggestion
token (PAT) which can be used e.g. in scripts.
```

> ## Token Creation

Reviewer: ```suggestion
## Personal access token creation
```

> ## Token Creation

Reviewer: ```suggestion
## PAT Creation
```
Let's call it PAT. Otherwise it's not clear whether we are talking about basic authentication tokens or Bearer tokens.

To create a Token you can go to Profile > Token and insert a short description
to create a token.

 <!-- prettier-ignore -->

!!! info The password which is generated can only be copied. Make sure you save

> it - you won't be able to access it again

Reviewer: This is broken: 
![image](https://github.com/DSD-DBS/capella-collab-manager/assets/23395732/831d68f6-0551-4cd0-846f-51f7ebafbb25)

## Token Scope

> Basic authentication token have the same scope as the bearer token.

Reviewer: ```suggestion
Basic authentication token have the same scope as the user who created it. It is therefore important that you never pass on the token and treat it responsibly. If you feel that the token has been lost, revoke it immediately and inform your Systems Engineering Toolchain team.
```

The typical user has no idea what a Bearer token is and wouldn't understand it.

<!-- prettier-ignore -->
!!! warning

>     As currently Basic Authentication token have the same scope as you logged in your browser they can act as your user. Please do not share the password and if the information gets lost please revoke the token as soon as possible.

Reviewer: Not needed anymore if you add the content of the last comment.

## Revoke a Token

In order to revoke a token go to Profile > Token. There you can see a list of
all tokens that belong to your user. Clicking on the trash symbol you can
delete a token which will no longer be valid to authenticate.

## Token Usage

The token created is a basic authentication token. There are different ways to
authenticate with that against the Collaboration Manager API. One example is:

```zsh

> curl --basic -u yourUsername:yourPassword https://baseurl/api/v1/users/current/tokens

Reviewer: ```suggestion
curl --basic -u <username>:<token> https://<baseURL>/api/v1/users/current/tokens
```

> curl --basic -u yourUsername:yourPassword https://baseurl/api/v1/users/current/tokens

Reviewer: This just reminded me that there is no public API documentation yet, added a new issue for it: https://github.com/DSD-DBS/capella-collab-manager/issues/1066. 

But this is out of scope for this PR.

```

> or to work with the diagram cache

Reviewer: Please include a bit more information. What is capellambse? The context is missing. 
Also add a link to the capellambse Github repository.

```python
capellambse.model.MelodyModel(
          path="path to the model on your machine",
          diagram_cache={

>             "path": "https://baseurl/api/projects/[yourProjectSlug]/[yourModelName]/diagrams/%s",

Reviewer: ```suggestion
            "path": "https://<baseURL>/api/projects/[yourProjectSlug]/[yourModelSlug]/diagrams/%s",
```

>             "username": "yourUsername",

Reviewer: ```suggestion
            "username": "<username>",
```

>             "password": "yourPassword",

Reviewer: ```suggestion
            "password": "<token>",
```

> capellambse.model.MelodyModel(
>           path="path to the model on your machine",
>           diagram_cache={
>             "path": "https://baseurl/api/projects/[yourProjectSlug]/[yourModelName]/diagrams/%s",
>             "username": "yourUsername",
>             "password": "yourPassword",
>           })

Reviewer: ```suggestion
capellambse.model.MelodyModel(
  path="path to the model on your machine",
  diagram_cache={
    "path": "https://baseurl/api/projects/[yourProjectSlug]/[yourModelName]/diagrams/%s",
    "username": "yourUsername",
    "password": "yourPassword",
  }
)
```

Wrong indentation and the closing bracket should be in a new line.
